packages/tools/viewer-alpha/src/viewerElement.ts

```diff
@@ -1,102 +1,455 @@
-import { Logger } from "core/Misc/logger";
-import type { Nullable } from "core/types";
+import type { PropertyValues } from "lit";
 import type { Viewer } from "./viewer";
+
+import { LitElement, css, html } from "lit";
+import { customElement, property, query, state } from "lit/decorators.js";
+
+import { Logger } from "core/Misc/logger";
 import { createViewerForCanvas } from "./viewerFactory";
 
-// TODO: Use https://lit.dev/ to simplify this code and ease maintenance.
+// Icon SVG is pulled from https://fluentuipr.z22.web.core.windows.net/heads/master/public-docsite-v9/storybook/iframe.html?id=icons-catalog--page&viewMode=story
+const playFilledIcon = "M17.22 8.68a1.5 1.5 0 0 1 0 2.63l-10 5.5A1.5 1.5 0 0 1 5 15.5v-11A1.5 1.5 0 0 1 7.22 3.2l10 5.5Z";
+const pauseFilledIcon = "M5 2a2 2 0 0 0-2 2v12c0 1.1.9 2 2 2h2a2 2 0 0 0 2-2V4a2 2 0 0 0-2-2H5Zm8 0a2 2 0 0 0-2 2v12c0 1.1.9 2 2 2h2a2 2 0 0 0 2-2V4a2 2 0 0 0-2-2h-2Z";
+
+const allowedAnimationSpeeds = [0.5, 1, 1.5, 2] as const;
+
+interface HTML3DElementEventMap extends HTMLElementEventMap {
+    modelchange: Event;
+    modelerror: Event;
+    selectedanimationchange: Event;
+    animationspeedchange: Event;
+    animationplayingchange: Event;
+    animationprogresschange: Event;
+}
 
 /**
  * Represents a custom element that displays a 3D model using the Babylon.js Viewer.
  */
-export class HTML3DElement extends HTMLElement {
-    // eslint-disable-next-line jsdoc/require-jsdoc, @typescript-eslint/naming-convention
-    public static readonly observedAttributes = Object.freeze(["src", "env"] as const);
-
+@customElement("babylon-viewer")

```

Comment:
Arent the decorators forcing a dependency on ts-lib ?

Reply:
I'm not actually sure, but since we already use decorators in @babylonjs/core is this a problem either way?

Reply:
good question, I know we have a part of tslib embedded in core to prevent the extra dependency, I do not know if it requires anything specific in the package even more if used outside core cc @RaananW ?

Reply:
I tried to read up on this a bit and from what I can tell, if you set `importHelpers` to `true` in `tsconfig.json`, then using certain TypeScript features will result in a dependency on `tslib`. If `importHelpers` is `false`, then it will inline the helper functions (like `__decorate` in the case of using decorators). Currently `importHelpers` is set to `true` in the root `tsconfig.json` in the repo, so we are picking this up for the alpha viewer just like every other package. I assume this won't be problematic in scenarios like Babylon Native since other Babylon packages are configured the same way. I'll double check with @RaananW when he is back. Also for the alpha viewer dist, everything is fully bundled, so the tslib helpers (e.g. `__decorate`) are included in the bundle.